### DISCUSSION TITLE
How is port forwarding port determined?

### DISCUSSION THREAD

**Discussion by alexjn1984** [qdm12/gluetun#2215] [Q&A]
I’ve been working with pfsense for a while and written some guides for users to connect to various providers with Wireguard.  What I am trying to figure out is how to determine the port assigned for port forwarding. 

I only know basic code so looking at the repo I couldn’t figure it out.  Any help would be greatly appreciated.

**Comment by zackherbert** [comment]
You can activate the [control server](https://github.com/qdm12/gluetun-wiki/blob/main/setup/advanced/control-server.md) by adding `-p 8000:8000/tcp` in your docker command.
Once that's done, you should get the forwarded port by running `curl http://localhost:8000/v1/openvpn/portforwarded`
